### PR TITLE
Add fashion mnist dataset

### PR DESCRIPTION
**Pull request by HeytemBou**
First draft for the new fashion-mnist dataset class

### CODE REVIEW

File: mplc/dataset.py [hunk 1]
```diff
@@ -613,3 +613,100 @@ def generate_new_model(self):
...
+    def preprocess_dataset_inputs(self, x):
+        x = x.astype("float32") / 255.0
+        return x.reshape(x.shape[0], self.img_width, self.img_height, 1)
+
+    def preprocess_dataset_labels(self, y):
+        return to_categorical(y, self.num_classes)
+
+    def generate_new_model(self):
+        """
+        Using a variation of the VGG16 model with batch normalization
```
Comment (bowni) [inline mplc/dataset.py:678]: @HeytemBou please indicate where this variation comes from: did you find it online? is it your creation?

**Review by bowni — CHANGES_REQUESTED** [review]

File: mplc/dataset.py [hunk 2]
```diff
@@ -17,9 +17,9 @@
 from librosa.feature import mfcc
 from loguru import logger
 from sklearn.model_selection import train_test_split
-from tensorflow.keras.datasets import cifar10, mnist, imdb
+from tensorflow.keras.datasets import cifar10, mnist, imdb, fashion_mnist
 from tensorflow.keras.layers import Activation
-from tensorflow.keras.layers import Conv2D, GlobalAveragePooling2D, MaxPooling2D
+from tensorflow.keras.layers import Conv2D, GlobalAveragePooling2D, MaxPooling2D, BatchNormalization
```
Comment (arthurPignet) [inline mplc/dataset.py:22]: Is batchNorm usable in federated context ? I asked myself the question back them but I did not really think about it, what do you think ?

File: mplc/dataset.py [hunk 3]
```diff
@@ -613,3 +613,100 @@ def generate_new_model(self):
...
+        model.add(BatchNormalization())
+        model.add(Dropout(0.5))
+
+        model.add(Dense(self.num_classes))
+        model.add(Activation("softmax"))
+
+        model.compile(
+            loss=categorical_crossentropy,
+            optimizer="adam",
+            metrics=self.model_metrics_names[1:],
```
Comment (arthurPignet) [inline mplc/dataset.py:710]: Did you test this model on a single partner learning way ? What accuracy did you achieve ?

File: mplc/dataset.py (hunk 1 above)
Comment (HeytemBou) [reply mplc/dataset.py:678]: @bowni I'll add the link in my next commit

File: mplc/dataset.py (hunk 2 above)
Comment (HeytemBou) [reply mplc/dataset.py:22]: Is there a particular reason for it to not be usable in the context of FL? There's not much in the literature that addresses the impact of BatchNorm on the final aggregated model, at least from what i've seen.

File: mplc/dataset.py (hunk 3 above)
Comment (HeytemBou) [reply mplc/dataset.py:711]: The model achieved about 93% test accuracy

File: mplc/dataset.py (hunk 2 above)
Comment (arthurPignet) [reply mplc/dataset.py:22]: I wonder what is used for the BN weights of the global model, especially at the end for testing.
I am not sure either about what happen at the aggregation step
What do you think ?

**Review by bowni — APPROVED** [review]

**Review by arthurPignet — APPROVED** [review]

**Review by RomainGoussault — APPROVED** [review]
Nice work @HeytemBou.
If this dataset is fast to run we should add a test. By fast I mean it should not add more than 10% of total test time.
